src/counter.tsx

```diff
@@ -0,0 +1,214 @@
+import * as styles from './counter.css';
+import {ThemeVariant, useThemeVariant} from './theme-variant-context';
+import {vars} from './skins/skin-contract.css';
+import Inline from './inline';
+import {Text3} from './text';
+import Touchable from './touchable';
+import IconSubtractRegular from './generated/mistica-icons/icon-subtract-regular';
+import IconAddMoreRegular from './generated/mistica-icons/icon-add-more-regular';
+import IconTrashCanRegular from './generated/mistica-icons/icon-trash-can-regular';
+import {useTheme} from './hooks';
+import classNames from 'classnames';
+import ScreenReaderOnly from './screen-reader-only';
+
+import type {DataAttributes} from './utils/types';
+
+const ICON_SIZE = 20;
+
+const useCounterState = ({
+    value,
+    defaultValue,
+    min,
+    max,
+    onChange,
+}: {
+    value?: number;
+    defaultValue?: number;
+    min: number;
+    max: number;
+    onChange?: (value: number) => void;
+}): [number, (value: number) => void] => {
+    const isControlledByParent = value !== undefined;
+
+    const getValueInRange = React.useCallback(
+        (value?: number) => {
+            return value === undefined ? min : Math.max(min, Math.min(max, value));
+        },
+        [min, max]
+    );
+
+    const [currentValue, setCurrentValue] = React.useState<number>(getValueInRange(defaultValue));
+
+    React.useEffect(() => setCurrentValue(getValueInRange(defaultValue)), [defaultValue, getValueInRange]);
+
+    const updateValue = (newValue: number) => {
+        if (!isControlledByParent) {
+            setCurrentValue(newValue);
+        }
+        if (onChange) {
+            onChange(newValue);
+        }
+    };
+
+    return [isControlledByParent ? getValueInRange(value) : currentValue, updateValue];
+};
+
+type Props = {
+    value?: number;
+    defaultValue?: number;
+    min?: number;
+    max?: number;
+    onChange?: (value: number) => void;
+    onRemove?: () => void;
+    children?: void;
+    disabled?: boolean;
+    dataAttributes?: DataAttributes;
+    removeLabel?: string;
+    increaseLabel?: string;
+    decreaseLabel?: string;
+    valueLabel?: string;
+};
+
+const Counter: React.FC<Props> = ({
+    value,
+    defaultValue,
+    onChange,
+    onRemove,
+    min,
+    max,
+    dataAttributes,
+    disabled,
+    removeLabel,
+    increaseLabel,
+    decreaseLabel,
+    valueLabel,
+}) => {
+    const variant = useThemeVariant();
+    const {texts} = useTheme();
+
+    const minValue = min === undefined ? 0 : min;
+    const maxValue = Math.max(minValue, max === undefined ? 999 : max);
+
+    const [currentValue, setCurrentValue] = useCounterState({
+        value,
+        defaultValue,
+        min: minValue,
+        max: maxValue,
+        onChange,
+    });
+
+    const hasTrashIcon = !!onRemove && currentValue === minValue;
+
+    const getRemoveLabel = () => {
+        return removeLabel === undefined ? texts.counterRemoveLabel : removeLabel;
```

Comment:
```suggestion
        return removeLabel  ?? texts.counterRemoveLabel;
```
(same for other labels)

Reply:
fixed